linchpin/__init__.py

```diff
@@ -288,10 +289,26 @@ def drop(ctx, pinfile, targets):
 
     pass
 
+@runcli.command()
+@click.argument('fetch_type', default=None, required=True)
+@click.argument('remote', default=None, required=True)
+@click.option('-r','--root', required=False, default=None)
+@pass_context
+def fetch(ctx, fetch_type, remote, root):
+    """
+    Fetches a aspecified linchpin directory from a remote location.
 
+    Fetch types: topology, layout, resources, hooks, workspace
 
-
-
+    """
+    lpcli = LinchpinCli(ctx)
+    lpcli.lp_fetch(remote, fetch_type, root)
+#    try:
```

Comment:
If you look at the lp_up, and lp_destroy commands, they both handle LinchpinError exceptions and call sys.exit as appropriate. This function should as well.